lib/Elastica/Index.php

```diff
@@ -100,23 +89,32 @@ public function getRecovery()
     }
 
     /**
-     * Gets all the type mappings for an index.
+     * Sets the mappings for the current index.
+     *
+     * @param Mapping $mapping MappingType object
+     * @param array   $query   querystring when put mapping (for example update_all_types)
+     *
+     * @return Response
+     */
+    public function setMapping(Mapping $mapping, array $query = []): Response
+    {
+        return $mapping->send($this, $query);
+    }
+
+    /**
+     * Gets all mappings for the current index.
      *
      * @return array
      */
-    public function getMapping()
+    public function getMapping(): array
     {
-        $response = $this->requestEndpoint(new Get());
+        $response = $this->requestEndpoint(new MappingGet());
         $data = $response->getData();
 
         // Get first entry as if index is an Alias, the name of the mapping is the real name and not alias name
         $mapping = \array_shift($data);
 
-        if (isset($mapping['mappings'])) {
-            return $mapping['mappings'];
-        }
-
-        return [];
+        return $mapping['mappings'] ?? [];
```

Comment:
Is this manual cleanup or some cleanup your IDE does automatically?

Reply:
My IDE highlights the code and suggests some optimizations :)